src/hooks/useContactProfile/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { useSession } from "@inrupt/solid-ui-react";
+import useSWR from "swr";
+import { asUrl } from "@inrupt/solid-client";
+import { getResource } from "../../solidClientHelpers/resource";
+import { getWebIdUrl } from "../../addressBook";
+import { fetchProfile } from "../../solidClientHelpers/profile";
+
+export default function useContactProfile(contact) {
+  const { fetch } = useSession();
+  let personThingUrl;
```

Comment:
Again, let's move the uses of `let` into specific functions. The use of try catch is also a code smell IMO, but I think it's a viable use in this case. But move it into a separate function to make the code easier to read.

Reply:
I used it only because I have no other way of finding out if a Thing is local or not, but will remove once that is available on solid client.